src/backend/distributed/sql/udfs/is_citus_depended_object/11.1-1.sql

```diff
@@ -0,0 +1,6 @@
+CREATE OR REPLACE FUNCTION pg_catalog.is_citus_depended_object(oid, oid)
```

Comment:
hmmm, do we really need these version files after we move the creation of the UDF to the tests? It'd be awesome to avoid. We really do not need any version control for a testing UDF.

For example, this is another UDF that is defined under src/test: https://github.com/citusdata/citus/blob/57455dc64dba521514c3bd85b2aab7f3cb2eecf8/src/test/regress/sql/single_node.sql#L843-L847


The trick is that you need to do `CREATE OR REPLACE FUNCTION ...` to enable that. Could we somehow squeeze ` `CREATE OR REPLACE FUNCTION ...` ` 

We have this very old hack where we create functions like that: https://github.com/citusdata/citus/blob/eb3e5ee227b5919a778bdacac6ff8046adffa0a5/src/test/regress/pg_regress_multi.pl#L933-L939

Can you try if we can do that for this function as well?

Reply:
**Info1:** When we run pg_regress --load-extension=citus ..., it creates a temp installation dir, drops the regression db if exist and creates it from scratch and load extensions given by --load-extension flag. It does all that stuff in drop_database and create_database functions in pg_regress.c file.  It also does a couple of alter database localization setups there.

**Info2:** To be able to do  the trick you mentioned above, we need to add the flag --use-existing to the pg_regress so that it can use our current pg installation directory. When use-existing is enabled, pg_regress does not execute drop_database and create_database functions, so --load-extension flag is ignored.

**Solution**
I can run tests successfully if I manually create db regression and create extension citus before pg_regress --use-existing except 2 tests. Those tests fail because of missing alter database localization setup done in create_database which would normally be called if I donot use --use-existing.

So, one option is to execute required commands specified in create_database function in pg_regress.c similarly in our script in a function and comment out why we do that.  Do you think that can be a valid solution?

Reply:
```perl
    # create default db
    system(catfile($bindir, "psql"),
            ('-X', '-h', $host, '-p', $masterPort, '-U', $user, "-d", "postgres",
                '-c', "CREATE DATABASE regression;")) == 0
            or die "Could not create regression database on master";
   
    # alter default db's lc_monetary
    system(catfile($bindir, "psql"),
            ('-X', '-h', $host, '-p', $masterPort, '-U', $user, "-d", "postgres",
                '-c', "ALTER DATABASE regression SET lc_monetary TO 'C';")) == 0
            or die "Could not alter regression database on master";

    # create extension citus
    system(catfile($bindir, "psql"),
            ('-X', '-h', $host, '-p', $masterPort, '-U', $user, "-d", "regression",
                '-c', "CREATE EXTENSION citus;")) == 0
            or die "Could not create citus extension on master";

    # we do not want to expose that udf other than vanilla tests
    my $citus_depended_object_def = "CREATE OR REPLACE FUNCTION pg_catalog.is_citus_depended_object(oid,oid) RETURNS bool LANGUAGE C AS 'citus', \$\$is_citus_depended_object\$\$";
    system(catfile($bindir, "psql"),
                    ('-X', '-h', $host, '-p', $masterPort, '-U', $user, "-d", "regression",
                    '-c', $citus_depended_object_def)) == 0
                or die "Could not create FUNCTION is_citus_depended_object on master";
```